src/wallet/walletdb.cpp

```diff
@@ -1314,6 +1314,29 @@ DBErrors WalletBatch::ZapSelectTx(std::vector<uint256>& vTxHashIn, std::vector<u
     return DBErrors::LOAD_OK;
 }
 
+bool RunWithinTxn(WalletDatabase& database, std::string_view process_desc, const std::function<bool(WalletBatch&)>& func)
+{
+    WalletBatch batch(database);
+    if (!batch.TxnBegin()) {
+        LogPrint(BCLog::WALLETDB, "Error: cannot create db txn for %s\n", process_desc);
+        return false;
+    }
+
+    // Run procedure, errors are logged internally
+    if (!func(batch)) {
+        batch.TxnAbort();
```

Comment:
In commit "wallet: db, introduce 'RunWithinTxn()' helper function" (b6dcfca8f77f32cb1f41230fb71bab6666252041)

I think it's bad for debugging that this function inconsistently logs errors when an operation fails. For example it seems like there is no error if the new eraserecords function fails? It would be better to add `LogPrint(BCLog::WALLETDB, "Error: %s failed\n", process_desc);` here.

I understand that in the future there may be cases where we may want to abort without logging errors but I think a better way to handle those cases would be to accept `kernel::Interrupted` return values here and use the `kernel::IsInterrupted()` helper. We can move these from `kernel` to `util` without breaking compatibility.

The comment above on line 1325 "errors are logged internally" is also not accurate and impossible to enforce with an anonymous callback, so would suggest dropping that and logging unconditionally here. (Without logging, the most the comment could say would be "errors need to be logged func callback" or "errors should be logged func callback")

Reply:
Yeah, using `util::Interrupted` to skip the failure logging in the future sounds great. Error logging added.